client/post-editor/controller.js

```diff
@@ -72,6 +75,39 @@ function maybeRedirect( context ) {
 		}
 	}
 
+	const isAtomic = isSiteAutomatedTransfer( state, siteId );
+	const isPrivate = isPrivateSite( state, siteId );
+	if ( isAtomic && isPrivate ) {
+		let currentEditor = get( context.query, 'set-editor', null );
+		if ( ! currentEditor ) {
+			currentEditor = getSelectedEditor( state, siteId );
+		}
+		if ( ! currentEditor ) {
+			await waitForSiteIdAndSelectedEditor( context );
+			currentEditor = getSelectedEditor( state, siteId );
+		}
```

Comment:
What's happening here? 

Can we combine the contents of the two `if ( ! currentEditor ) {` blocks?

Reply:
We only want to `await` if `context.query` and `getSelectedEditor` selector returned a falsy value. I'll simplify it a little bit.